smart/IGM.py

```diff
@@ -0,0 +1,116 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import skimage as ski
+from skimage.morphology import disk
+
+import astropy.units as u
+
+from sunpy.map import Map
+
+from smart.map_processing import STL
+
+
+def IGM(im_map: Map, diff_map: Map, dilation_radius: u.Quantity[u.arcsec] = 2.5 * u.arcsec):
```

Comment:
Few small things functions usually are lower case unless it's an acronym which this this but but the current name isn't very desriptive, also the argument names could be better so what about something like

```
def grow_mask_and_index(current_map, previous_map ...
```